be/src/column/array_column.cpp

```diff
@@ -407,6 +407,19 @@ Datum ArrayColumn::get(size_t idx) const {
     return Datum(res);
 }
 
+void ArrayColumn::iterate_values(size_t idx, const std::function<void(const Datum&, int)>& cb_value,
+                                 const std::function<void(size_t)>& cb_size) const {
+    DCHECK_LT(idx + 1, _offsets->size()) << "idx + 1 should be less than offsets size";
+    size_t offset = _offsets->get_data()[idx];
+    size_t array_size = _offsets->get_data()[idx + 1] - offset;
+
+    for (int i = 0; i < array_size; ++i) {
+        cb_value(_elements->get(offset + i), i);
+    }
+
+    cb_size(array_size);
```

Comment:
I think you could get the offsets and elements columns firstly. then do this work in retention function.